src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisher.java

```diff
@@ -61,30 +65,48 @@ public void publish(final ChecksDetails details) {
             GitHubAppCredentials credentials = context.getCredentials();
             GitHub gitHub = Connector.connect(StringUtils.defaultIfBlank(credentials.getApiUri(), gitHubUrl),
                     credentials);
-
-            GitHubChecksDetails gitHubDetails = new GitHubChecksDetails(details);
-
-            Optional<Long> existingId = context.getId(gitHubDetails.getName());
-
-            final GHCheckRun run;
-
-            if (existingId.isPresent()) {
-                run = getUpdater(gitHub, gitHubDetails, existingId.get()).create();
+            boolean skipPublish = false;
+            String contributor = context.getContributor();
+            buildLogger.log("contributor name is " + contributor);
+            if (contributor.isEmpty()) {
+                skipPublish = false;
+            } else {
+                String repository = context.getRepository();
+                boolean isPrivate = gitHub.getRepository(repository).isPrivate();
+                GitHubStatusChecksProperties gitHubStatusChecksProperties = new GitHubStatusChecksProperties();
+                boolean publishNonConfluentIncPR = gitHubStatusChecksProperties.isPublishNonConfluentIncPR(context.getJob());
+                boolean publishConfluentIncPR = gitHubStatusChecksProperties.isPublishConfluentIncPR(context.getJob());
+                buildLogger.log("publishNonConfluentIncPR is " + publishNonConfluentIncPR);
+                buildLogger.log("publishConfluentIncPR is " + publishConfluentIncPR);
+                boolean isConfluentInc = orgCheck(contributor, gitHub, isPrivate);
+                skipPublish = skipPublish(isPrivate, isConfluentInc, publishNonConfluentIncPR, publishConfluentIncPR);
+                buildLogger.log("skip publishing github status : " + skipPublish);
             }
-            else {
-                run = getCreator(gitHub, gitHubDetails).create();
+            if (!skipPublish) {
+                GitHubChecksDetails gitHubDetails = new GitHubChecksDetails(details);
```

Comment:
are the contents of this `then` block the same as the original method?

Reply:
yes